.github/workflows/main.yml

```diff
@@ -0,0 +1,20 @@
+on: [push, pull_request]
```

Comment:
This could be changed to:
```yml
on:
  push:
    # Run against master branch only 
    branches: [master]
  pull_request:
    # Include event types emitted on code change only
    types: [opened, synchronize]
```